### PR TITLE
gist website embed link

### PR DESCRIPTION
**Pull request by tkgdsg**
You can paste GitBucket Gist Code in your website.

please review this patch.

### CODE REVIEW

File: src/main/twirl/gitbucket/gist/detail.scala.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  revision: String,
+  files: Seq[(String, String)]
+  )(implicit context: gitbucket.core.controller.Context)
+@import gitbucket.core.view.helpers
+@import play.twirl.api.Html
+@bootstrapCSS={<link href="@helpers.assets("/vendors/bootstrap-3.3.6/css/bootstrap.css")" rel="stylesheet">}
+@prettifyCSS={<link href="@helpers.assets("/vendors/google-code-prettify/prettify.css")" rel="stylesheet">}
+@gistCSS={<link href="@context.path/plugin-assets/gist/style.css" rel="stylesheet">}
+@gitbucketCSS={<link href="@helpers.assets("/common/css/gitbucket.css")" rel="stylesheet">}
+@adminLTECSS={<link href="@helpers.assets("/vendors/AdminLTE-2.3.8/css/AdminLTE.min.css")" rel="stylesheet">}
```
Comment (takezoe) [inline src/main/twirl/gitbucket/gist/detail.scala.js]: Why do you define these variables even through you are using them only once?

File: src/main/scala/gitbucket/gist/controller/GistController.scala [hunk 2]
```diff
@@ -415,7 +417,12 @@ trait GistControllerBase extends ControllerBase {
   ////////////////////////////////////////////////////////////////////////////////
 
 
-  private def _gist(userName: String, repoName: Option[String] = None, revision: String = "master"): Any = {
+  private def _gist(userName: String, repoName: Option[String] = None, revision: String = "master", isEmbed: Boolean = false): Any = {
```
Comment (takezoe) [inline src/main/scala/gitbucket/gist/controller/GistController.scala]: Who pass `true` to `isEmbed` parameter?

Comment (tkgdsg) [inline src/main/scala/gitbucket/gist/controller/GistController.scala]: I think I found better solution.
I changed function structure. Please review new one

File: src/main/scala/gitbucket/gist/controller/GistController.scala [hunk 3]
```diff
@@ -415,7 +417,12 @@ trait GistControllerBase extends ControllerBase {
...
+
+    if( repoName.isDefined && repoName.get.endsWith(".js")) {
+      return _gist(userName, Some(repoName.get.substring(0,repoName.get.length()-3)), revision, true)
```
Comment (tkgdsg) [inline src/main/scala/gitbucket/gist/controller/GistController.scala]: I think I found better solution.
I changed function structure. Please review new one

File: src/main/twirl/gitbucket/gist/detail.scala.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+@(gist: gitbucket.gist.model.Gist,
+  repositoryUrl: gitbucket.gist.util.GistUtils.GistRepositoryURL,
+  revision: String,
+  files: Seq[(String, String)]
+  )(implicit context: gitbucket.core.controller.Context)
+@import gitbucket.core.view.helpers
+@import play.twirl.api.Html
```
Comment (tkgdsg) [inline src/main/twirl/gitbucket/gist/detail.scala.js]: I removed unnecessary import statement.

File: src/main/twirl/gitbucket/gist/detail.scala.js (hunk 1 above)
Comment (tkgdsg) [inline src/main/twirl/gitbucket/gist/detail.scala.js]: I modified that code to use strings directly in "document.write()".

**Comment by tkgdsg** [comment]
I modified codes in response to the review.

**Comment by takezoe** [comment]
@tkgdsg Looks good. Great thanks for this pull request!

**Pull request merged**